homeassistant/components/fritz/quality_scale.yaml

```diff
@@ -0,0 +1,96 @@
+rules:
+  # Bronze
+  action-setup:
+    status: todo
+    comment: still in async_setup_entry, needs to be moved to async_setup
+  appropriate-polling: done
+  brands: done
+  common-modules: done
+  config-flow-test-coverage:
+    status: todo
+    comment: one coverage miss in line 110
+  config-flow: done
+  dependency-transparency: done
+  docs-actions: done
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions:
+    status: todo
+    comment: include the proper docs snippet
+  entity-event-setup: done
+  entity-unique-id: done
+  has-entity-name:
+    status: todo
+    comment: partially done
+  runtime-data:
+    status: todo
+    comment: still uses hass.data
+  test-before-configure: done
+  test-before-setup: done
+  unique-config-entry: done
+
+  # Silver
+  action-exceptions: done
+  config-entry-unloading: done
+  docs-configuration-parameters: done
+  docs-installation-parameters:
+    status: todo
+    comment: add the proper configuration_basic block
+  entity-unavailable: done
+  integration-owner: done
+  log-when-unavailable: done
+  parallel-updates:
+    status: todo
+    comment: not set at the moment, we use a coordinator
+  reauthentication-flow: done
+  test-coverage:
+    status: todo
+    comment: we are at 88% coverage at the moment
+
+  # Gold
+  devices: deno
+  diagnostics: done
+  discovery-update-info: todo
+  discovery: done
+  docs-data-update: todo
+  docs-examples: done
+  docs-known-limitations:
+    status: exempt
+    comment: no known limitations, yet
+  docs-supported-devices:
+    status: todo
+    comment: add the known supported devices
+  docs-supported-functions:
+    status: todo
+    comment: need to be overhauled
+  docs-troubleshooting: done
+  docs-use-cases:
+    status: todo
+    comment: need to be overhauled
+  dynamic-devices: done
+  entity-category: done
+  entity-device-class: done
+  entity-disabled-by-default: done
+  entity-translations: done
+  exception-translations: done
+  icon-translations: done
+  reconfiguration-flow: done
+  repair-issues:
+    status: exempt
+    comment: no known use cases for repair issues or flows, yet
+  stale-devices:
+    status: todo
+    comment: implement async_remove_config_entry_device in addition to the cleanup service
```

Comment:
Why add the function instead of automating it?

Reply:
Depending on the selected discovery method, we can not be 100% sure, when a tracked device is not available, if it is on purpose or not - therefor we had implemented the cleanup service/button in the past

Reply:
Yes, but in theory, the clean up service and button are abstractions on top of HA so nowadays we wouldn't really allow it I think (not sure if this is a shared opinion or that I am missing some history there)

Reply:
ok - so we should replace the button/service by some automatic method?

Reply:
Ideally, but like you said, you can't be 100% sure, so maybe that is something we should dig out to see what's the reason of that (maybe it has been dug out before)

Reply:
we had the same discussion in #116024 with the outcome to implement the automated way. I think this would be the better approach, so i'll change the todo comment accordingly

Reply:
```suggestion
    comment: automate the current cleanup process and deprecate the corresponding button
```